Revit_Core_Engine/Modify/CopyParameters.cs

```diff
@@ -129,6 +129,12 @@ public static void CopyParameters(this Element element, IBHoMObject bHoMObject,
                 // Sort the parameters so that doubles get assigned last, to make sure all reference levels etc. go first.
                 foreach (RevitParameter param in fragment.Parameters.Where(x => !(x.Value is double)).Union(fragment.Parameters.Where(x => x.Value is double)))
                 {
+                    if (param.IsReadOnly)
+                    {
+                        BH.Engine.Base.Compute.RecordError($"'{param.Name}' is a read-only Revit parameter so cannot be modified.");
```

Comment:
I would turn this into a warning - overall action will work, and this is just one piece of it (or at least this is my gut feeling here)